node/full_node_integration_test.go

```diff
@@ -344,6 +344,87 @@ func testSingleAggreatorSingleFullNodeSingleLightNode(t *testing.T) {
 	assert.Equal(n1h, n3h, "heights must match")
 }
 
+func testSingleAggreatorSingleFullNodeFraudProofGossip(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	var wg sync.WaitGroup
+	aggCtx, aggCancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(context.Background())
+	clientNodes := 1
+	nodes, _ := createNodes(aggCtx, ctx, clientNodes+1, true, &wg, t)
+
+	node1 := nodes[0]
+	node2 := nodes[1]
+
+	require.NoError(node1.Start())
+	time.Sleep(2 * time.Second)
+	require.NoError(node2.Start())
+
+	time.Sleep(2 * time.Second)
+
+	n1Frauds, err := node1.fraudService.Get(aggCtx, types.StateFraudProofType)
+	require.NoError(err)
+	aggCancel()
+	require.NoError(node1.Stop())
+
+	n2Frauds, err := node2.fraudService.Get(aggCtx, types.StateFraudProofType)
+	require.NoError(err)
+	cancel()
+	require.NoError(node2.Stop())
+
+	assert.Greater(len(n1Frauds), 0, "number of fraud proofs gossiped should be greater than 0")
```

Comment:
Why is the malicious aggregator gossiping state fraud proofs? It's the one committing the fraud

Reply:
`n1Frauds` corresponds to number of fraud proofs received via gossip and since the aggregator is part of the p2p network, it does receive the fraud proofs that are gossiped.